hack/tools/release/notes/print.go

```diff
@@ -167,6 +171,19 @@ REPLACE ME: A couple sentences describing the deprecation, including links to do
 		}
 	}
 
+	// Print the list of dependencies
+	fmt.Println("")
+
+	repoURL := fmt.Sprintf("https://github.com/%s", p.repo)
+	deps, err := notes.NewDependencies().ChangesForURL(
+		repoURL, p.fromTag, p.toTag,
+	)
```

Comment:
I would move this call out of the printer, just to not pollute its responsibilities
Maybe change the input to be a struct containing the entries and these deps